### PR TITLE
[Process] Deprecate ProcessBuilder

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by fabpot — APPROVED** [review]

File: src/Symfony/Component/Process/ProcessBuilder.php [hunk 1]
```diff
@@ -11,13 +11,17 @@
 
 namespace Symfony\Component\Process;
 
+@trigger_error(sprintf('The %s class is deprecated since version 3.4 and will be removed in 4.0. Use the Process class instead.', __METHOD__), ProcessBuilder::class);
```
Comment (pierredup) [inline src/Symfony/Component/Process/ProcessBuilder.php]: The arguments at the end needs to be `, ProcessBuilder::class), E_USER_DEPRECATED);`

Comment (nicolas-grekas) [inline src/Symfony/Component/Process/ProcessBuilder.php]: thanks, fixed

**Review by romainneutron — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by javiereguiluz** [comment]
Please, whenever you deprecate something, add a small comment explaining the main reasons. Thanks!

**Comment by stof** [comment]
The reason here is that the main use case for the ProcessBuilder was to escape the individual parts of the command for you. And this is now supported by the Process class itself by passing an array as the command (where it will escape each element of the array and build the final string command). The ProcessBuilder even relies on this feature now.

**Pull request merged**